fontforge/scripting.c

```diff
@@ -265,15 +266,17 @@ static void calldatafree(Context *c) {
     /* child may have freed some args itself by shifting,
        but argc will reflect the proper values none the less */
     for ( i=1; i<c->a.argc; ++i ) {
-        if ( c->a.vals[i].type == v_str ) {
-            free( c->a.vals[i].u.sval );
-            c->a.vals[i].u.sval = NULL;
-        }
-	if ( c->a.vals[i].type == v_arrfree || (c->a.vals[i].type == v_arr && c->dontfree[i]!=c->a.vals[i].u.aval )) {
+	if ( c->a.vals[i].flags & vf_dontfree )
+	    c->a.vals[i].flags ^= vf_dontfree;
```

Comment:
Why is it necessary to unset the flag?

Reply:
Ah, I see. It's the sort of thing that would normally fit a reference count pattern, but I guess that that would represent an unnecessarily large change from the old logic.

Reply:
Yeah, I actually tried refcounting first, but it's a bit of a headache to get right, because none of the current usages assume that, which makes it difficult exactly where to increment/decrement the counts.